### DISCUSSION TITLE
How to retrieve data only up to open of current trading session?

### DISCUSSION THREAD

**Discussion by nicholashoyy** [ib-api-reloaded/ib_async#64] [Q&A] [answered]
Hello,

I am trying to retrieve data using ib.reqHistoricalData.  And I want to make reference to open price of the current trading session (instead of calendar day) for instruments which trade almost around the clock e.g. ES, GC, etc.  I am thinking of using pandas loc and locate row 0.

However, durationStr = 1D returns 1 calendar day worth of bars.  Are there ways to work around this?

Thank you.

Nicholas

**Comment by gnzsnz** [comment, marked as answer]
this looks more like "how do i get the current trading session?" question.
I use [exchange calendars](https://github.com/gerrymanoim/exchange_calendars) or eventually [historical schedule](https://ib-api-reloaded.github.io/ib_async/api.html#ib_async.ib.IB.reqHistoricalSchedule)

to get the open price, i usually do

```python
ticker = ib.reqMktData(contract)
ib.sleep(1)
ticker.open
```
which will only work if the session is open

hope it helps
